doozer/doozerlib/backend/rebaser.py

```diff
@@ -954,12 +969,40 @@ def _add_build_repos(self, dfp: DockerfileParser, metadata: ImageMetadata, dest_
             all_stages=True,
         )
 
+        last_stage = self.split_dockerfile_into_stages(dfp)[-1]
+
+        # Find all the USERs in the last stage
+        user_stanzas = []
+        for line in last_stage:
+            if "USER" in line.keys():
+                user_stanzas.append(f"USER {line['USER']}")
+
+        # The Dockerfile should have only one unique non-USER 0 value
+        if "USER 0" in user_stanzas:
+            # list.remove() command removes values inline
+            user_stanzas.remove("USER 0")
+        if user_stanzas and len(user_stanzas) != 1:
```

Comment:
A stage can have any number of users. Not sure why the check.
```
USER 1
RUN make some files for this user
USER 2 
RUN make some file for this user
etc.
```

Reply:
updated in https://github.com/openshift-eng/art-tools/pull/1452/commits/f40809e03f9cad43ab1b6bf40583d729189280ee